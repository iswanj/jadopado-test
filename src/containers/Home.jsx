import React, { PureComponent } from 'react';
import styles from 'home.css';
import { connect } from 'react-redux';
import { setState } from '../actions';

import BottomNavigation from './components/BottomNavigation';

export default class Home extends PureComponent {
  render() {
    return (
      <div className={styles.mainContainer}>
        <section className={styles.products}>
          <div className={styles.item}>
            <div className={styles.topHeader}>
              <div className={styles.user}>
                <img alt="User Pic" src="https://randomuser.me/api/portraits/men/13.jpg" className={styles.userPic} />
                <span className={styles.username}>tracymacgrady</span>
              </div>
              <div className={styles.distance}>
                <span>500m</span> away
              </div>
            </div>
          </div>
        </section>
        <BottomNavigation />
      </div>
    );
  }
}

// Connect to Redux store
function mapStateToProps(state, ownProps) {
  return {

  };
}

export const HomeContainer = connect(
  mapStateToProps,
  { setState }
)(Home);
