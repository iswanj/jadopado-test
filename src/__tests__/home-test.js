jest.mock('react-dom');
import React from 'react';
import Home from '../containers/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Home page renders correctly', () => {
  const products = [
    {
      "id": 1,
      "title": "Leaf iPhone Case Hard Plastic",
      "owner": {
        "username": "tracymacgrady",
        "image": "https://randomuser.me/api/portraits/men/13.jpg"
      },
      "images": [
        "leaf-iphone-case.jpg"
      ],
      "price": 450,
      "distance": 500,
      "likes": 32,
      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
      "tages": ["iphone", "cases", "mobile_phones", "macbookpro"],
      "comments": [
        {
          "id": 1,
          "username": "jevonjans",
          "image": "https://randomuser.me/api/portraits/men/98.jpg",
          "comment": "Adhuc liberavisse cu pro. Ex eam mutat ancillae noluisse. Ius ad autem aliquid vulputate"
        },
        {
          "id": 2,
          "username": "robynrihanna",
          "image": "https://randomuser.me/api/portraits/women/28.jpg",
          "comment": "Te nec soleat graece commune, cu pri ferri verear dignissim, case wisi ad vim. Modo iusto aperiam eu eos, etiam convenire assueverit cum no."
        }
      ]
    }
  ];
  const tree = renderer.create(
    <Home products={products} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
