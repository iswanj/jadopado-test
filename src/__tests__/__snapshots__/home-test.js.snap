exports[`test Home page renders correctly 1`] = `
<div
  className={undefined}>
  <section
    className={undefined}>
    <div
      className={undefined}>
      <div
        className={undefined}>
        <div
          className={undefined}>
          <img
            alt="User Pic"
            className={undefined}
            src="https://randomuser.me/api/portraits/men/13.jpg" />
          <span
            className={undefined}>
            tracymacgrady
          </span>
        </div>
        <div
          className={undefined}>
          <span>
            500
            m
          </span>
           away
        </div>
      </div>
      <div
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "width": "1024px",
          }
        }>
        <div
          className={undefined}
          style={
            Object {
              "backgroundImage": "url(/images/leaf-iphone-case.jpg)",
              "width": "1024px",
            }
          } />
        <div
          className={undefined}>
          <div
            className={undefined}>
            <p>
              Leaf iPhone Case Hard Plastic
            </p>
            <p>
              <strong>
                AED 
                450
              </strong>
            </p>
          </div>
          <div
            className={undefined}>
            <i
              className="  " />
            <i
              className="  " />
          </div>
        </div>
      </div>
      <div
        className={undefined}>
        <i
          className="  " />
         
        32
         likes
      </div>
      <div
        className={undefined}>
        <p
          className={undefined}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit
          <span
            className={undefined}>
             
            #iphone 
            #cases 
            #mobile_phones 
            #macbookpro 
          </span>
        </p>
        <a
          className={undefined}>
          View 
          2
           comments
        </a>
      </div>
    </div>
  </section>
  <div
    className={undefined}>
    <a
      className={undefined}
      onClick={[Function]}
      style={Object {}}>
      <i
        className="  " />
    </a>
    <a
      className={undefined}
      onClick={[Function]}
      style={Object {}}>
      <i
        className="  " />
    </a>
    <a
      className={undefined}
      onClick={[Function]}
      style={Object {}}>
      <i
        className="  " />
    </a>
    <a
      className={undefined}
      onClick={[Function]}
      style={Object {}}>
      <i
        className="  " />
    </a>
    <p
      className=" "
      onClick={[Function]}>
      <i
        className="  " />
    </p>
  </div>
  <div
    className=" false">
    <div
      className={undefined}>
      <input
        placeholder="What are you looking for?"
        type="text" />
    </div>
    <div
      className={undefined}>
      <p>
        Search within 
        <strong>
          0
          km
        </strong>
      </p>
      <div
        className={undefined}>
        <div
          aria-disabled={false}
          className="InputRange"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}>
          <span
            className="InputRange-label InputRange-label--min">
            <span
              className="InputRange-labelContainer">
              0
            </span>
          </span>
          <div
            className="InputRange-track InputRange-track--container"
            onMouseDown={[Function]}
            onTouchStart={[Function]}>
            <div
              className="InputRange-track InputRange-track--active"
              style={
                Object {
                  "left": "0%",
                  "width": "0%",
                }
              } />
            <span
              className="InputRange-sliderContainer"
              style={
                Object {
                  "left": "0%",
                  "position": "absolute",
                }
              }>
              <span
                className="InputRange-label InputRange-label--value">
                <span
                  className="InputRange-labelContainer">
                  0
                </span>
              </span>
              <a
                aria-controls={undefined}
                aria-labelledby={undefined}
                aria-valuemax={120}
                aria-valuemin={0}
                aria-valuenow="0"
                className="InputRange-slider"
                draggable="false"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="slider" />
            </span>
          </div>
          <span
            className="InputRange-label InputRange-label--max">
            <span
              className="InputRange-labelContainer">
              120
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className={undefined}>
      <div
        className={undefined}>
        <h2>
          trending search
        </h2>
        <p
          className={undefined}>
          <span>
            #hammock
          </span>
           
          <span>
            #iceberg
          </span>
           
          <span>
            #whenindubai
          </span>
           
          <span>
            #lettuceleaf
          </span>
           
          <span>
            #Isxv8
          </span>
          <span>
            #marvel
          </span>
           
          <span>
            #dc
          </span>
           
          <span>
            #cat
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
